docs/reference/components/sources/dnstap.cue

```diff
@@ -0,0 +1,1325 @@
+				required:      false
+				type: object: {
+					options: {
+						time: {
+							description: """
+								The time at which the DNS query/update request message
+								was sent or received. This is the number of time units
+								(determined by 'timePrecision') since the UNIX epoch.
+								"""
+							required: true
+							type: uint: {
+								unit: null
+								examples: [1614781642516276825]
+							}
+						}
+						timePrecision: {
+							description: "The time precision used by field 'time'."
+							required:    true
+							type: string: {
+								enum: {
+									s:  "second"
+									ms: "millisecond"
+									us: "microsecond"
+									ns: "nanosecond"
+								}
+								syntax: "literal"
+							}
+						}
+						fullRcode: {
+							common: true
+							description: """
+								The numeric rcode that is the sum of the 4bits header's
+								rcode + the 8bits opt's extendedRcode when present.
+								Should be 0 for request.
+								"""
+							required: false
+							type: uint: {
+								unit: null
+								examples: [0]
+								default: null
+							}
+						}
+						rcodeName: {
+							common: true
+							description: """
+								Textual response code corresponding to the 'fullRcode'.
+								Should be 'No Error' for request.
+								"""
+							required: false
+							type: string: {
+								enum: {
+									NoError:   "No Error"
+									FormErr:   "Format Error"
+									ServFail:  "Server Failure"
+									NXDomain:  "Non-Existent Domain"
+									NotImp:    "Not Implemented"
+									Refused:   "Query Refused"
+									YXDomain:  "Name Exists when it should not"
+									YXRRSet:   "RR Set Exists when it should not"
+									NXRRSet:   "RR Set that should exist does not"
+									NotAuth:   "Server Not Authoritative for zone"
+									NotZone:   "Name not contained in zone"
+									BADSIG:    "TSIG Signature Failure"
+									BADKEY:    "Key not recognized"
+									BADTIME:   "Signature out of time window"
+									BADMODE:   "Bad TKEY Mode"
+									BADNAME:   "Duplicate key name"
+									BADALG:    "Algorithm not supported"
+									BADTRUNC:  "Bad Truncation"
+									BADCOOKIE: "Bad/missing server cookie"
+								}
+								default: null
+								syntax:  "literal"
+							}
+						}
+						rawData: {
+							common: false
+							description: """
+								Raw binary request message data encoded in Base64.
+								Presents only upon failures.
+								"""
+							required: false
+							type: string: {
+								examples: ["YoWEAAABAAAAAQAAAmgyB2V4YW1wbGUDY29tAAAGAAHADwAGAAEAAA4QAD0LYmRkcy1kbnN0YXAACnBvc3RtYXN0ZXICbm8FZW1haWwGcGxlYXNlACQAxj4AAA4QAAACWAAnjQAAAA4Q"]
+								default: null
+								syntax:  "literal"
+							}
+						}
+						header: {
+							common: true
+							description: """
+								Header section of DNS query/update request message.
+								See DNS related RFCs for detailed information about
```

Comment:
Usually, users who want to see dnstap data should have been quite familiar with those RFCs already. But, yeah, since you've raised it, I'll try to enumerate them explicitly.

Reply:
Updated.